src/lib/transition_handler/processor.ml

```diff
@@ -279,16 +286,17 @@ let run ~logger ~(precomputed_values : Precomputed_values.t) ~verifier
                             subtree
                             (* It could be the case that by the time we try and
                                * add the breadcrumb, it's no longer relevant when
-                               * we're catching up *)
-                            ~f:
-                              (add_and_finalize ~logger ~only_if_present:true
-                                 ~source:`Catchup ~valid_cb:None ) )
+                               * we're catching up *) ~f:(fun (b, valid_cb) ->
+                              add_and_finalize ~logger ~only_if_present:true
+                                ~source:`Catchup ~valid_cb b ) )
                     with
                   | Ok () ->
                       ()
                   | Error err ->
                       List.iter breadcrumb_subtrees ~f:(fun tree ->
-                          Rose_tree.iter tree ~f:(fun cached_breadcrumb ->
+                          Rose_tree.iter tree
+                            ~f:(fun (cached_breadcrumb, _vc) ->
+                              (* TODO reject callback? *)
```

Comment:
An error here doesn't indicate an error. It's probably fine to just drop the validation callback in this rather rare edge case, though technically we should ignore any validations which have not yet been filled.